embassy-nxp/src/usart/lpc55.rs

```diff
@@ -0,0 +1,854 @@
+use core::marker::PhantomData;
+use embassy_hal_internal::{Peri, PeripheralType};
+use embedded_io::{self, ErrorKind};
+pub use sealed::SealedInstance;
+use crate::gpio::AnyPin;
+
+
+/// Serial error
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[non_exhaustive]
+pub enum Error {
+    /// Triggered when the FIFO (or shift-register) is overflowed.
+    Overrun,
+    /// Triggered when a break is received
+    Break,
+    /// Triggered when there is a parity mismatch between what's received and
+    /// our settings.
+    Parity,
+    /// Triggered when the received character didn't have a valid stop bit.
+    Framing,
+    /// Triggered when the receiver detects noise?
+    Noise,
+}
+
+
+impl embedded_io::Error for Error {
+    fn kind(&self) -> ErrorKind {
+        match self {
+            Error::Overrun => ErrorKind::Other,
+            Error::Parity => ErrorKind::InvalidData,
+            Error::Framing => ErrorKind::InvalidData,
+            Error::Break => ErrorKind::Interrupted,
+            Error::Noise => ErrorKind::Other,
+        }
+    }
+}
+/// Word length.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum DataBits {
+    /// 7 bits.
+    #[doc = "7 bit Data length."]
+    DataBits7,
+    /// 8 bits.
+    #[doc = "8 bit Data length."]
+    DataBits8,
+    /// 9 bits.
+    #[doc = "9 bit data length. The 9th bit is commonly used for addressing in multidrop mode."]
+    DataBits9,
+}
+
+impl DataBits {
+    fn bits(&self) -> u8 {
+        match self {
+            Self::DataBits7 => 0b00,
+            Self::DataBits8 => 0b01,
+            Self::DataBits9 => 0b10,
+        }
+    }
+}
+
+/// Parity bit.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum Parity {
+    /// No parity.
+    #[doc = "No parity."]
+    ParityNone,
+    /// Even parity.
+    #[doc = "Even parity."]
+    ParityEven,
+    /// Odd parity.
+    #[doc = "Odd parity."]
+    ParityOdd,
+}
+
+impl Parity {
+    fn bits(&self) -> u8 {
+        match self {
+            Self::ParityNone => 0b00,
+            Self::ParityEven => 0b10,
+            Self::ParityOdd => 0b11,
+        }
+    }
+}
+
+/// Stop bits.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum StopBits {
+    #[doc = "1 stop bit."]
+    STOP1,
+    #[doc = "2 stop bits. This setting should only be used for asynchronous communication."]
+    STOP2,
+}
+
+impl StopBits {
+    fn bits(&self) -> bool {
+        return match self {
+            Self::STOP1 => false,
+            Self::STOP2 => true,
+        };
+    }
+}
+
+/// UART config.
+#[non_exhaustive]
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
```

Comment:
nitpick: IMHO, `Copy`, `PartialEq` and `Eq` doesn't make sense to me. While we can implement them on it, it might bloat the debug builds with extra code. And we already know how painfully slow flashing the LPC is :(

```suggestion
#[derive(Clone,  Debug)]
```